### PR TITLE
Add instruction for code coverage checks

### PR DESCRIPTION
**Pull request by maytasm**
Add instruction for code coverage checks

### Description

Add instruction for code coverage checks to make Druid more user friendly for Devs.

This PR has:
- [x] been self-reviewed.
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

File: dev/intellij-setup.md [hunk 1]
```diff
@@ -34,6 +34,14 @@ an alias name. You can do this in Using `File` -> `Project Structure...` -> `Pla
...
 The Code Style is available in XML format at [druid_intellij_formatting.xml](druid_intellij_formatting.xml) and can be [imported into IntelliJ](https://www.jetbrains.com/help/idea/2017.1/copying-code-style-settings.html).
 
+## Set Code Coverage Runner
+Druid CI checks are configured to enforce code coverage using JaCoCo. The checks will prevent PR from being merged 
+if test coverage of new added code is below the set threshold. You should run the tests locally to make sure that
+your code pass the coverage threshold. In IntelliJ, edit the template of JUnit tests in "Run Configurations..." 
+to use Jacoco for code coverage (this provides branch coverage instead of the default which is only line coverage).
+You can then right click src/test/java folder of the modules you are modifying and click run with coverage. This
+will generate a report to show the current code coverage on the code (not just your change).
+
```
Comment (suneet-s) [inline dev/intellij-setup.md:44]: Can you add screenshots to this section to make it easier for devs to follow along

File: dev/code-review/code-coverage.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+
+# Druid's Code Coverage Enforcement
+
+Druid has an automated way of checking if new code has adequate unit tests.
```
Comment (suneet-s) [inline dev/code-review/code-coverage.md:22]: I think we should call out that our current code coverage checks are merely smoke tests.

They only verify that a line/ branch of code has been called during the test, but not that the functionality has been tested sufficiently. Reviewers should still verify that all the different branches are sufficiently tested by reviewing the tests.

File: dev/code-review/code-coverage.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Druid CI checks are configured to enforce code coverage using JaCoCo. The CI checks will prevent PR from being merged 
+if test coverage of new added code is below the set threshold. The CI checks filters test coverage based on a diff from
+your PR and make sure that the thresholds are met. Druid currently enforce branch and line code coverage.
+
+## Running code coverage locally
+Code coverage should be run locally to make sure the PR will pass Druid CI checks. 
+1. Code coverage on the codebase can be generated directly in [Intellij](../intellij-setup.md#Set-Code-Coverage-Runner). 
+2. Code coverage on just the diff of your PR can be generated in your terminal. First, you will have to install
+diff-test-coverage by running `npm install @connectis/diff-test-coverage`. Next, run the unit tests
+for the module you are working on `mvn -pl <MODULE_TO_CHECK> test jacoco:report`. Finally, run
```
Comment (ccaominh) [inline dev/code-review/code-coverage.md:32]: We can also mention that this creates an HTML report in `target/site/jacoco/index.html`.

File: dev/intellij-setup.md [hunk 4]
```diff
@@ -34,6 +34,14 @@ an alias name. You can do this in Using `File` -> `Project Structure...` -> `Pla
 ## Code Style
 The Code Style is available in XML format at [druid_intellij_formatting.xml](druid_intellij_formatting.xml) and can be [imported into IntelliJ](https://www.jetbrains.com/help/idea/2017.1/copying-code-style-settings.html).
 
+## Set Code Coverage Runner
+Druid CI checks are configured to enforce code coverage using JaCoCo. The checks will prevent PR from being merged
```
Comment (ccaominh) [inline dev/intellij-setup.md:38]: typo: prevent RR -> prevent a PR

File: dev/code-review/code-coverage.md (hunk 2 above)
Comment (maytasm) [reply dev/code-review/code-coverage.md:22]: Done

File: dev/intellij-setup.md (hunk 1 above)
Comment (maytasm) [reply dev/intellij-setup.md:44]: Done

File: dev/code-review/code-coverage.md (hunk 3 above)
Comment (maytasm) [reply dev/code-review/code-coverage.md:32]: Done

File: dev/intellij-setup.md (hunk 4 above)
Comment (maytasm) [reply dev/intellij-setup.md:38]: done

**Pull request merged**